### DISCUSSION TITLE
mpremote on smartphone/tablet termux

### DISCUSSION THREAD

**Discussion by picpic020960** [micropython/micropython#10446] [General]
Bonjour ,
Try to run mpremote on smartphone.
The USB otg string is /dev/bus/usb/001/054 with termux ide (termux-usb -l)
Google said me TCPUART (play store) and socat 
( socat pty,link=$HOME/ttyesp32,raw TCP: < IP from TCPUART > : < PORT from TCPUART >
Not easy : mpremote/pyserial not accept USB otg string format : this above and the alias /dev/pts/3 )
One trick simple and 'durable' ?

Thanks for help
Regards

**Comment by jimmo** [comment]
I haven't tested this myself, but mpremote needs a path to a tty device. it sounds like `mpremote connect $HOME/ttyesp32` might work? But it definitely won't work with either the /dev/bus or /dev/pts devices.

But in general, mpremote in termux isn't supported or tested. Contributions welcome though!
